### DISCUSSION TITLE
Aqara (Xiaomi) RTCGQ14LM - OTA updates not working?

### DISCUSSION THREAD

**Discussion by francismiles1** [Koenkk/zigbee2mqtt#16707] [Devices]
Hi
I own 3 of the [P1 motion sensors](https://www.zigbee2mqtt.io/devices/RTCGQ14LM.html) 
In general, they work, although really oddly (they don't trigger if you approach direct,ly etc)
But, my question is about OTA firmware updates for these devices. The zigbee2mqtt page tells me it's supported.

Two of these devices are on a slightly older firmware version (0.0.0_0005), the newer one is 0.0.0_0006
I would like to update these two, but the OTA feature doesn't seem to be working correctly. 
Is this something I need to configure correctly? Not supported? Special mode I need to set these devices in?
Its interesting, the devices report {"installed_version":-1,"latest_version":-1,"state":null} in all payload responses.
Would like to know either way, just for my sanity!

So, HomeAssistant, the latest everything. 
Using ZZH! stick, CC2652R_coordinator_20221226 firmware.

> Debug 2023-02-16 18:11:16Received MQTT message on 'zigbee2mqtt/bridge/request/device/ota_update/check' with data '{"id":"Aqara Motion Sensor Lounge","transaction":"l2s0f-1"}'
> Info 2023-02-16 18:11:16Checking if update available for 'Aqara Motion Sensor Lounge'
> Debug 2023-02-16 18:11:16Check if update available for '0x54ef4410004e464b' (lumi.motion.ac02)
> Debug 2023-02-16 18:11:16Using endpoint '1'
> Info 2023-02-16 18:12:16MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Aqara Motion Sensor Lounge"},"error":"Failed to check if update available for 'Aqara Motion Sensor Lounge' (Device didn't respond to OTA request)","status":"error","transaction":"l2s0f-1"}'
> Error 2023-02-16 18:12:16Failed to check if update available for 'Aqara Motion Sensor Lounge' (Device didn't respond to OTA request)
> Debug 2023-02-16 18:12:16Error: Device didn't respond to OTA request at requestOTA (/app/node_modules/zigbee-herdsman-converters/lib/ota/common.js:191:15) at Object.isUpdateAvailable (/app/node_modules/zigbee-herdsman-converters/lib/ota/common.js:246:25) at OTAUpdate.onMQTTMessage (/app/lib/extension/otaUpdate.ts:203:45)
> Debug 2023-02-16 18:12:40Received Zigbee message from 'Aqara Motion Sensor Passage', type 'attributeReport', cluster 'aqaraOpple', data '{"illuminance":65582}' from endpoint 1 with groupID 0
> Debug 2023-02-16 18:12:40lumi.motion.ac02: Processed data into payload {}
> Info 2023-02-16 18:12:40MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Passage', payload '{"battery":100,"detection_interval":30,"device_temperature":21,"illuminance":41,"linkquality":127,"motion_sensitivity":"high","occupancy":true,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":3010}'
> Debug 2023-02-16 18:12:40Received Zigbee message from 'Aqara Motion Sensor Lounge', type 'attributeReport', cluster 'aqaraOpple', data '{"illuminance":65588}' from endpoint 1 with groupID 0
> Debug 2023-02-16 18:12:40lumi.motion.ac02: Processed data into payload {}
> Info 2023-02-16 18:12:40MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Lounge', payload '{"battery":100,"detection_interval":30,"device_temperature":21,"illuminance":47,"linkquality":127,"motion_sensitivity":"medium","occupancy":true,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":3028}'

Thanks!

**Comment by Benkei** [comment]
After some trial and error.
My device was on version 0.0.0_0006 and updated to 0.0.0_0010.

- press pairing button
- start OTA update
- check the Logs, wait for the message "Starting upgrade" (Log level: Debug)
> debug 2023-03-17 21:02:33: Starting upgrade
- press pairing button
- now the update process should start and in Logs you can watch the process
> debug 2023-03-17 21:02:46: OTA update at 0%, remaining Infinity seconds
> debug 2023-03-17 21:03:17: OTA update at 2.4%, remaining 1763.2147511697149 seconds
> debug 2023-03-17 21:23:27: OTA update at 98.06%, remaining 24.79723768643143 seconds
> debug 2023-03-17 21:23:53: Update succeeded, waiting for device announce
> ...

good luck

**Reply by stncttr908** [reply]
When you say "press pairing button" do you mean briefly press and release, or do you mean hold until the blue light flashes, indicating it's in pairing/repairing mode? Thanks!

**Reply by francismiles1** [reply]
I tried both methods. Didn't work, unfortunately.

**Reply by lux73** [reply]
> When you say "press pairing button" do you mean briefly press and release, or do you mean hold until the blue light flashes, indicating it's in pairing/repairing mode? Thanks!

only short press - if you press & hold until blue flash the Decive goes into pairing mode then

@Benkei

thx, works like a charm as described - have updated both of my P1 Sensors few minutes ago from 0.0.0_0006 to latest 0.0.0_0010 👍🏻 

btw keep in mind that updating battery powered devices is a heavy battery draining process - both of my P1 had 100% (3000mV+) before and after updating the battery there is only 33% (2900mV) left

**Reply by stncttr908** [reply]
Finally got around to this, worked great! My battery also shows exactly 33% now though so sounds like a bug to me. 

**Update** never mind, battery shows fine now

**Comment by hserna74** [comment]
For me P1 the last version available is 0.0.0_0006.

I tried several times to upgrade, but always the system display there is not new version

**Reply by lux73** [reply]
do you using plain z2m or ZHA Addon?

/me is using plain z2m

**Reply by hserna74** [reply]
Home Assistant Add-on: Zigbee2MQTT

**Reply by lux73** [reply]
then it seems the ZHA Addon does not have full functionality - i would advise switch to plain z2m

**Comment by hserna74** [comment]
Ok. Noob question, how I change to? Url or video that explain proc?

**Comment by francismiles1** [comment]
Am wondering if OTA needs specific port access. I have tried all sorts of configurations, but always get the following payload response:
`payload '{"battery":87,"detection_interval":30,"device_temperature":21,"illuminance":214,"linkquality":138,"motion_sensitivity":"medium","occupancy":true,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":2980}`

Note the -1 for version

**Reply by lubomirp** [reply]
Hi, i have the same payload for RTCGQ14LM, was you able to fix that issue with OTA update? I have found, that firmware update is possible via connecting P1 to the Aqara Hub :/

**Reply by francismiles1** [reply]
Unfortunately not. It seems to detect there is an update but everytime I try, it fails.
```
Press Button
Info 2023-11-01 12:08:25	Checking if update available for 'Aqara Motion Sensor Lounge'
Info 2023-11-01 12:08:29	MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Lounge', payload '{"battery":71,"detection_interval":30,<<Lots_of_info_here>>,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":2956}'
Info 2023-11-01 12:08:30	Update available for 'Aqara Motion Sensor Lounge'
Info 2023-11-01 12:08:30	MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Lounge', payload '{"battery":71,"detection_interval":30,<<Lots_of_info_here>>,"update":{"installed_version":5,"latest_version":10,"state":"available"},"update_available":null,"voltage":2956}'
Info 2023-11-01 12:08:30	MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Aqara Motion Sensor Lounge","updateAvailable":true},"status":"ok","transaction":"ljw3v-2"}'
Info 2023-11-01 12:08:30	MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Lounge', payload '{"battery":71,"detection_interval":30,"device_temperature":21,"illuminance":73,"linkquality":112,"motion_sensitivity":"medium","occupancy":false,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":5,"latest_version":10,"state":"available"},"update_available":null,"voltage":2956}'
Info 2023-11-01 12:09:22	Updating 'Aqara Motion Sensor Lounge' to latest firmware
Info 2023-11-01 12:10:45	MQTT publish: topic 'zigbee2mqtt/Aqara Motion Sensor Lounge', payload '{"battery":71,"detection_interval":30,<<Lots_of_info_here>>,"update":{"installed_version":5,"latest_version":10,"state":"available"},"update_available":null,"voltage":2956}'
Info 2023-11-01 12:10:45	MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"id":"Aqara Motion Sensor Lounge"},"error":"Update of 'Aqara Motion Sensor Lounge' failed (Device didn't respond to OTA request)","status":"error","transaction":"ljw3v-3"}'
Error 2023-11-01 12:10:45	Update of 'Aqara Motion Sensor Lounge' failed (Device didn't respond to OTA request)
```

==EDIT== Just fiddled again. It's working now (On ONE unit Only, The other two refuse to update).. I got the button push sequence wrong
it should be as follows:

1. in zigbee2mqtt ->OTA -> Click on Update Firmware (or check for new updates)
2. THEN press the little button on the back briefly - You should see the blue LED flash for a brief time

I was doing it in the wrong sequence!!

![image](https://github.com/Koenkk/zigbee2mqtt/assets/28559506/f5c432d5-7bf0-4c76-ae21-b094ff57019c)

**Reply by lubomirp** [reply]
congratulation :)

there must be some special magic, i am trying it like you mentioned above, exactly:
1. click on button check for new updates in z2m gui
2. press the little button for approx. 0,5 sec and then release the button - LED flashes (blue)

But the error "Failed to find endpoint which support OTA cluster" appears immediately after the i clicked on the button "check for new updates", so the click on the little button on the back is always to late (maybe 0.3sec after after i clicked the button).

<img width="319" alt="Screenshot 2023-11-01 at 16 07 03" src="https://github.com/Koenkk/zigbee2mqtt/assets/96984461/02dcf451-0eb5-4bbf-8f34-09b6764a2f0f">

Just to be sure, you have just clicked on the button "check for new updates" in z2m gui, and afterwards briefly pressed and released the little button on the back of P1 and nothing more?


```
**here i pressed the button for approx. 0.5sec and the released, LED has flashed (blue)**
Debug 2023-11-01 16:18:02Received Zigbee message from 'motionsensor', type 'attributeReport', cluster 'aqaraOpple', data '{"illuminance":65549}' from endpoint 1 with groupID 0
Info 2023-11-01 16:18:02MQTT publish: topic 'zigbee2mqtt/motionsensor', payload '{"battery":100,"detection_interval":20,"device_temperature":26,"illuminance":13,"linkquality":228,"motion_sensitivity":"medium","occupancy":true,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":3144}'
Debug 2023-11-01 16:18:08Received MQTT message on 'zigbee2mqtt/bridge/request/device/ota_update/check' with data '{"id":"motionsensor","transaction":"2u4wk-222"}'
Info 2023-11-01 16:18:08Checking if update available for 'motionsensor'
Info 2023-11-01 16:18:08MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Checking if update available for 'motionsensor'","meta":{"device":"motionsensor","status":"checking_if_available"},"type":"ota_update"}'
Info 2023-11-01 16:18:08MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Failed to check if update available for 'motionsensor' (Failed to find endpoint which support OTA cluster)","meta":{"device":"motionsensor","status":"check_failed"},"type":"ota_update"}'
Info 2023-11-01 16:18:08MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"motionsensor"},"error":"Failed to check if update available for 'motionsensor' (Failed to find endpoint which support OTA cluster)","status":"error","transaction":"2u4wk-222"}'
Error 2023-11-01 16:18:08Failed to check if update available for 'motionsensor' (Failed to find endpoint which support OTA cluster)
Debug 2023-11-01 16:18:09Received Zigbee message from 'motionsensor', type 'attributeReport', cluster 'genBasic', data '{"modelId":"lumi.motion.ac02"}' from endpoint 1 with groupID 0
Debug 2023-11-01 16:18:15Received Zigbee message from 'motionsensor', type 'attributeReport', cluster 'aqaraOpple', data '{"247":{"data":[1,33,74,12,3,40,26,4,33,0,0,5,33,1,0,8,33,10,1,10,33,0,0,12,32,1,19,32,0,20,32,0,100,16,0,101,33,14,0,105,32,20,106,32,2,107,32,0],"type":"Buffer"}}' from endpoint 1 with groupID 0
Info 2023-11-01 16:18:15MQTT publish: topic 'zigbee2mqtt/motionsensor', payload '{"battery":100,"detection_interval":20,"device_temperature":26,"illuminance":14,"linkquality":216,"motion_sensitivity":"medium","occupancy":true,"power_outage_count":0,"trigger_indicator":false,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":3146}'
```

UPDATE: i have tried to update fw in my Zigbee 3.0 USB Dongle Plus V2 to 6.10.3, but still the same.

**Reply by francismiles1** [reply]
'Special Magic' is correct.. This only happened on one of my three devices. I still can't get the other two devices to 'wake up' to OTA!! They both show -1 as the software version ("update":{"installed_version":-1,"latest_version":-1,"state":null})
Perhaps its got something to do with the direction of the wind ;-)

**Comment by yandy-r** [comment]
I have one of these sensors that I attempted to do an OTA upgrade. It seemed to work, and went to 100%, but the reported firmware version doesn't change. Then when I check for OTA update, z2m responds with no updates found.

Reported version is 0.0.0_0006 and the one it originally found and seemed to update to is 0.0.0_0010, though it just keeps reporting the pervious one.

Wondering if anyone has seen this.

**Comment by hserna74** [comment]
Equals than me.

I have got the same problem

**Reply by hserna74** [reply]
The status info displays

    "update": {
        "installed_version": 10,
        "latest_version": 10,
        "state": "idle"
    },

About information displays

Firmware date
Jun 9 2022
Firmware Versión
0.0.0_0006

**Reply by Sjekke** [reply]
I know ... old thread ... but does anyone resolved this?

**Reply by francismiles1** [reply]
Seems not. Still cant update my other two devices.
